### PR TITLE
rmlint 2.4.4 (new formula)

### PR DESCRIPTION
**Pull request by tomhoover**

### CODE REVIEW

**Comment by MikeMcQuaid** [comment]
@BrewTestBot test this please

File: Formula/rmlint.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  url "https://github.com/sahib/rmlint/archive/v2.4.4.tar.gz"
+  sha256 "294708e7c98783a7782df1ed7f6fc79e9036571b7f69f76c5b3455545ce568bc"
+
+  depends_on "glib" => :run
+  depends_on "scons" => :build        # build system
+  depends_on "gettext" => :build      # support for localization
+  depends_on "pkg-config" => :build   # manage compile and link flags for libraries
+  depends_on "sphinx-doc" => :build   # manpage/documentation generation - Build manpage from docs/rmlint.1.rst
+  depends_on "libelf" => :optional    # find non-stripped binaries
+  depends_on "json-glib" => :optional # support for reading json caches
```
Comment (MikeMcQuaid) [inline Formula/rmlint.rb]: Only need these comments for `:optional` and they are better handled with e.g. `option "with-json-glob", "Add support for reading JSON caches.'`

File: Formula/rmlint.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+  def install
+    scons "config"
+    scons "-j 4"
+    bin.install "rmlint"
```
Comment (MikeMcQuaid) [inline Formula/rmlint.rb]: Is there no `make install` equivalent we can use?

File: Formula/rmlint.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  depends_on "scons" => :build        # build system
+  depends_on "gettext" => :build      # support for localization
+  depends_on "pkg-config" => :build   # manage compile and link flags for libraries
+  depends_on "sphinx-doc" => :build   # manpage/documentation generation - Build manpage from docs/rmlint.1.rst
+  depends_on "libelf" => :optional    # find non-stripped binaries
+  depends_on "json-glib" => :optional # support for reading json caches
+
+  def install
+    scons "config"
+    scons "-j 4"
```
Comment (MikeMcQuaid) [inline Formula/rmlint.rb]: Remove the `-j 4`, pretty sure we set it sensibly by default and you don't know how many cores a given user machine will have.

File: Formula/rmlint.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    bin.install "rmlint"
+    man1.install "docs/rmlint.1.gz"
+  end
+
+  test do
+    (testpath/"1.txt").write("1")
+    (testpath/"2.txt").write("2")
+    system bin/"rmlint"
```
Comment (MikeMcQuaid) [inline Formula/rmlint.rb]: Nice test. Is it worth creating a dupe and verifying it deletes that?

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]
Looking good so far 👍

File: Formula/rmlint.rb (hunk 1 above)
Comment (tomhoover) [inline Formula/rmlint.rb]: done

File: Formula/rmlint.rb (hunk 3 above)
Comment (tomhoover) [inline Formula/rmlint.rb]: done

**Comment by tomhoover** [comment]
@MikeMcQuaid, everything should be ready for another look.

File: Formula/rmlint.rb [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    scons "config"
+    scons
+    bin.install "rmlint"
+    man1.install "docs/rmlint.1.gz"
+  end
+
+  test do
+    (testpath/"1.txt").write("1")
+    (testpath/"2.txt").write("1")
+    system "(#{bin}/rmlint | grep '^# Duplicate(s):$') || false"
```
Comment (MikeMcQuaid) [inline Formula/rmlint.rb]: Final tweak. Instead do:

```ruby
assert_match(/^# Duplicate(s):$/, shell_output("#{bin}/rmlint")
```

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

File: Formula/rmlint.rb (hunk 5 above)
Comment (tomhoover) [inline Formula/rmlint.rb]: @MikeMcQuaid, final tweak incorporated.

Comment (tomhoover) [inline Formula/rmlint.rb]: @MikeMcQuaid, nevermind--the suggested change failed the build (whereas the previous test passed). I'll take a look at it again this evening.

Comment (MikeMcQuaid) [inline Formula/rmlint.rb]: Looking again: `assert_match(/^# Duplicate\(s\):$/, shell_output("#{bin}/rmlint").strip` may do it correctly or just the looser `assert_match "# Duplicate(s):", shell_output("#{bin}/rmlint")`

**Comment by tomhoover** [comment]
@MikeMcQuaid, I didn't have access to run tests from the command line while at work today, and attempted to "loosen" the search criteria by editing the file using a browser. It was over 8 hours before the test was re-run, but it failed again. Once got back home to my computer, I was able to review the build log in detail and found the following syntax will work:

`assert_match "# Duplicate(s):", shell_output("#{bin}/rmlint")`.

This was my first attempt at authoring a homebrew formula from scratch, so the assert_match was something new to me. I'm always learning! 😄

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**